static/app/components/organizations/projectPageFilter.tsx

```diff
@@ -0,0 +1,493 @@
+          : selectedProjects.slice(0, 1);
+
+      const label = isAllProjectsSelected
+        ? t('All Projects')
+        : isMyProjectsSelected
+        ? t('My Projects')
+        : projectsToShow.map(proj => trimSlug(proj.slug, 25)).join(', ');
+
+      // Number of projects that aren't listed in the trigger label
+      const remainingCount = isAllProjectsSelected
+        ? 0
+        : isMyProjectsSelected
+        ? value.length - memberProjects.length
+        : value.length - projectsToShow.length;
+
+      return (
+        <DropdownButton
+          {...props}
+          icon={
+            !projectsLoaded ||
+            !pageFilterIsReady ||
+            isAllProjectsSelected ||
+            isMyProjectsSelected ? (
+              <IconProject />
+            ) : (
+              <PlatformList
+                platforms={projectsToShow.map(p => p.platform ?? 'other').reverse()}
+              />
+            )
+          }
+        >
+          <TriggerLabel>
+            {!projectsLoaded || !pageFilterIsReady ? t('Loading\u2026') : label}
+          </TriggerLabel>
+          {remainingCount > 0 && <StyledBadge text={`+${remainingCount}`} />}
+        </DropdownButton>
+      );
+    },
+    [value, memberProjects, nonMemberProjects, pageFilterIsReady, projectsLoaded]
+  );
+
+  const menuFooter = useMemo(() => {
+    const hasProjectWrite = organization.access.includes('project:write');
+    if (!hasProjectWrite) {
+      return null;
+    }
+
+    return (
+      <Fragment>
+        <Feature
+          organization={organization}
+          features={['organizations:global-views']}
+          hookName="feature-disabled:project-selector-all-projects"
+          renderDisabled={false}
+        >
+          {({renderShowAllButton}: FeatureRenderProps) => {
+            // if our hook is adding renderShowAllButton, render that
+            if (showNonMemberProjects && renderShowAllButton) {
+              return renderShowAllButton({
+                onButtonClick: () => handleChange([]),
+                canShowAllProjects: showNonMemberProjects,
+              });
+            }
+
+            return null;
+          }}
+        </Feature>
+        <Button
+          size="xs"
+          aria-label={t('Add Project')}
+          to={`/organizations/${organization.slug}/projects/new/`}
+          icon={<IconAdd size="xs" isCircled />}
+        >
+          {t('Project')}
+        </Button>
+      </Fragment>
+    );
+  }, [organization, handleChange, showNonMemberProjects]);
+
+  const menuWidth = useMemo(() => {
+    const flatOptions: SelectOption<number>[] = options.flatMap(item =>
+      'options' in item ? item.options : [item]
+    );
+
+    const longestSlug = flatOptions
+      .slice(0, 25)
+      .reduce(
+        (acc, cur) => (String(cur.label).length > acc ? String(cur.label).length : acc),
+        0
+      );
+
+    return `${Math.max(20, Math.min(28, 12 + longestSlug * 0.6))}em`;
```

Comment:
This looks like a lot of magic to me 😬 - maybe needs a comment?

Reply:
Got u https://github.com/getsentry/sentry/pull/47195/commits/5bdaa40d70cf0da5251e4cfc4f67b2b98a3a02e0